File: src/ru/gurps/generator/desktop/models/rules/Quirk.java
package ru.gurps.generator.desktop.models.rules;

import ru.gurps.generator.desktop.config.Model;

public class Quirk extends Model {
    public Integer id;
    public String name;
    @Ignore public Integer cost = 0;
    @Ignore public Boolean add = false;

    public Quirk() {
    }

    public Quirk(String language) {
        this.name = language;
    }

    public String getName() {
        return name;
    }

    public String getCost() {
        return Integer.toString(cost);
    }

    public Boolean getAdd() {
        return add;
    }
}

File: src/main/kotlin/ru/gurps/generator/desktop/models/rules/Quirk.kt
package ru.gurps.generator.desktop.models.rules

import ru.gurps.generator.desktop.config.Model
import java.sql.ResultSet

class Quirk : Model {
    override var id: Int = -1
    var name: String = ""
    var cost: Int = 0
    var costString: String
        get() = cost.toString()
        set(value) {
            cost = value.toInt()
        }
    var add: Boolean = false

    constructor()
    constructor(result: ResultSet) {
        id = result.getInt("id")
        val nameRes = result.getString("name")
        if (nameRes != null) name = nameRes
    }

    constructor(name: String) {
        this.name = name
    }
}
